design/sales.md

```diff
@@ -20,24 +19,142 @@ or being negotiated.
     |                           ^         ^           |
     ----------------------------|---------|-----------
                                 |         |
-                  availability  |         | state
+                 reserved space |         | state
                                 v         v
                    ----------------    -----------------
                    |     Repo     |    |   Datastore   |
                    ----------------    -----------------
 
-The Sales module keeps track of storage reservations in its internal
-Reservations module. It keeps a record of which pieces of reserved storage
-belong to which storage contract. It queries and updates the amount of available
-storage in the Repo. It uses a Datastore to persist its own state.
+The reservations module keeps track of storage that is available to be sold.
+Users are able to add availability to indicate how much storage they are willing
+to sell and under which conditions.
 
-The Repo exposes the following functions that allow the Reservations module to
-query and update the amount of available storage:
+    Availability
+      amount
+      maximum duration
+      minimum price
+
+Availabilities consist of an amount of storage, the maximum duration and minimum
+price to sell it for. They represent storage that is for sale, but not yet sold.
+This is information local to the node that can be altered without affecting
+global state.
+
+Adding availability
+-------------------
+
+When a user adds availability, then the reservations module wil check whether
+there is enough space available in the Repo. If there is enough space, then it
+will increase the amount of reserved space in the Repo. It persists the state of
+all availabilities to the Datastore, to ensure that they can be restored when a
+node is restarted.
+
+    User          Reservations          Repo      Datastore
+     |                  |                  |            |
+     | add availability |                  |            |
+     | ---------------->| check free space |            |
+     |                  |----------------->|            |
+     |                  | reserve amount   |            |
+     |                  |----------------->|            |
+     |                  |                               |
+     |                  | persist availability          |
+     |                  |------------------------------>|
+
+Selling storage
+---------------
+
+When a request for storage is submitted on chain, the sales module decides
+whether or not it wants to act on it. First, it tries to find an availability
+that matches the requested amount, duration, and price. The matching
+availability will be removed from the reservations module, so that it can't be
+sold twice. If an availability matches, but is larger than the requested
+storage, then the Sales module may decide to split the availability into a part
+that we can use for the request, and a remainder that can be sold separately.
```

Comment:
```suggestion
sold twice.
```
We may not really need this clarification. In the reservations module, we are simply releasing bytes are they are stored, and updating the used availability's size to reflect the remaining bytes. Once the bytes have been completely downloaded, the availability will have the correct, reduced, size.

Reply:
I'd consider that to be an implementation detail. The design allows for availability to be split, and we're free to implement that as we see fit.